app/views/projects/show.html.erb

```diff
@@ -21,7 +21,7 @@
     </thead>
     <tbody>
       <% if @project.stages.to_a.any? %>
-        <% cache [@project, 'stages', Lock.cache_key] do %>
+        <% cache [@project, 'stages', Lock.cache_key, @project.stages.map(&:updated_at).max] do %>
```

Comment:
does .map create a n+1? pluck or a query involving sql max perhaps?

Reply:
it fetches the stages in a single query ... so not really expensive ... 
we already do the `.to_a` above so the stages are preloaded